src/posh/lib/ratom.cljc

```diff
@@ -17,146 +17,147 @@
-  (get [])
-  (set [x])))
+   (definterface IMutable
+     (get [])
+     (set [x])))
 
 ; TODO CLJ create for necessary primitive datatypes as well
 #?(:clj
-(deftype Mutable [^:unsynchronized-mutable val]
-  IMutable
-  (get [this] val)
-  (set [this x] (set! val x))
-  IDeref
-  (deref [this] val)))
+   (deftype Mutable [^:unsynchronized-mutable val]
+     IMutable
+     (get [this] val)
+     (set [this x] (set! val x))
+     IDeref
+     (deref [this] val)))
 
 #?(:clj (defmacro mut [x] `(Mutable. ~x)))
 
 #?(:clj
-(defmacro getm
-  "Get mutable"
-  [x]
-  (if-cljs &env x
-                `(.get ~(with-meta x {:tag 'posh.lib.ratom.Mutable})))))
+   (defmacro getm
+     "Get mutable"
+     [x]
+     (if-cljs &env x
+                   `(.get ~(with-meta x {:tag 'posh.lib.ratom.Mutable})))))
 
 #?(:clj
-(defmacro setm!
-  "Set mutable"
-  [x v]
-  (if-cljs &env `(set!            ~x                                        ~v)
-                `(.set ~(with-meta x {:tag 'posh.lib.ratom.Mutable}) ~v))))
+   (defmacro setm!
+     "Set mutable"
+     [x v]
+     (if-cljs &env `(set!            ~x                                        ~v)
+                   `(.set ~(with-meta x {:tag 'posh.lib.ratom.Mutable}) ~v))))
 
 #?(:clj
-(defmacro getf
-  "Get field"
-  [x field]
-  (let [accessor (symbol
-                   (str "."
-                     (if-cljs &env (str "-" (name field))
-                                   (str "get" (upper-first (name field))))))]
-    `(~accessor ~x))))
+   (defmacro getf
+     "Get field"
+     [x field]
+     (let [accessor (symbol
+                      (str "."
+                        (if-cljs &env (str "-" (name field))
+                                      (str "get" (upper-first (name field))))))]
+       `(~accessor ~x))))
 
 #?(:clj
-(defmacro setf!
-  "Set field"
-  [x field v]
-  (let [accessor (symbol
-                   (str "."
-                     (if-cljs &env (str "-" (name field))
-                                   (str "set" (upper-first (name field))))))]
-    (if-cljs &env `(set! (~accessor ~x) ~v)
-                  `(~accessor ~x ~v)))))
+   (defmacro setf!
+     "Set field"
+     [x field v]
+     (let [accessor (symbol
+                      (str "."
+                        (if-cljs &env (str "-" (name field))
+                                      (str "set" (upper-first (name field))))))]
+       (if-cljs &env `(set! (~accessor ~x) ~v)
+                     `(~accessor ~x ~v)))))
 
 #?(:clj
-(defmacro add! [x v]
-  (if-cljs &env `(.push ~x ~v)
-                `(.add ~(with-meta x {:tag 'java.util.ArrayList}) ~v))))
+   (defmacro add! [x v]
+     (if-cljs &env `(.push ~x ~v)
+                   `(.add ~(with-meta x {:tag 'java.util.ArrayList}) ~v))))
 
 #?(:clj
-(defmacro array-list [& args]
-  (if-cljs &env `(array ~@args)
-                `(doto (ArrayList.) ~@(for [arg args] `(.add ~arg))))))
+   (defmacro array-list [& args]
+     (if-cljs &env `(into [] ~@args)
```

Comment:
This is going to cause issues with ‘alength’  and ‘aget’ further down, as well as a few other things. A JS array is extensible so it’s basically an arraylist. Persistent vectors’ properties are too dissimilar to just swap it in here.

Reply:
The array macro doesn't seem to exist in cljs anymore. What is the correct function to use then?

Reply:
Sorry, don't know why I didn't respond to this! Just wanted to say that `array` is definitely still there — see also https://github.com/clojure/clojurescript/blob/master/src/main/clojure/cljs/core.cljc#L2482.

Reply:
Though I think I might know why it's not compiling properly — try:
```
`(cljs.core/array ~@args)
```